avro/src/main/java/com/fasterxml/jackson/dataformat/avro/jsr310/deser/AvroLocalDateDeserializer.java

```diff
@@ -0,0 +1,53 @@
+package com.fasterxml.jackson.dataformat.avro.jsr310.deser;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+/**
+ * Deserializer for {@link LocalDate} from and integer value.
+ *
+ * Deserialized value represents number of days from the unix epoch, 1 January 1970.
+ *
+ * Deserialization from string is not supported.
+ */
+public class AvroLocalDateDeserializer extends StdScalarDeserializer<LocalDate> {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final AvroLocalDateDeserializer INSTANCE = new AvroLocalDateDeserializer();
+
+    protected AvroLocalDateDeserializer() {
+        super(LocalDate.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public LocalDate deserialize(JsonParser p, DeserializationContext context) throws IOException, JsonProcessingException {
```

Comment:
It seems to me that the pattern for deserialization is similar if not identical to all types: if `VALUE_NUMBER_INT` is gotten, get `long` value, call `fromLong()`; otherwise call `handleUnexpectedToken()`.
So maybe extract a shared base class that does `deserialize()` and leaves `fromLong()` as `abstract.

Further, need to implement

    public LogicalType logicalType() { return LogicalType.DateTime; }

(added in 2.12)

Reply:
Good hint,

- I have created base deserializer `AvroJavaTimeDeserializerBase` with abstract method `fromLong(long longValue, ZoneId defaultZoneId)` method.

    However `ZoneId  defaultZoneId` is not used by deserializers to timezone-less types (2 out of 4 deserializers). 
Hope it is OK and ` context.getTimeZone().toZoneId().normalized()` does not add performance significant penalty.

- `logicalType()` added.